wallet/concepts/wallet-api.md

```diff
@@ -58,26 +55,46 @@ The RPC methods are documented in the interactive [JSON-RPC API reference](/wall
 :::
 
+Dapps communicate with MetaMask through JSON-RPC methods. 
+These methods are divided into the following:
+
+- [Restricted methods](#restricted-methods) -  Require user consent for actions that impact assets or data (for example initiating a transaction).
+- [Unrestricted methods](#unrestricted-methods) - Allow dapps to perform basic actions without permission (for example retrieving a public address).
+
 ### Restricted methods
 
-MetaMask introduced wallet permissions in [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
-In this permissions system, each RPC method is restricted or unrestricted.
-If a method is restricted, a dapp must request permission to call it using
-[`wallet_requestPermissions`](/wallet/reference/wallet_requestpermissions).
+MetaMask implements permissions based on [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255) to enhance security for when users interact with dapps. 
+This requires that dapps obtain user consent before accessing certain features. 
 Under the hood, permissions are plain, JSON-compatible objects, with fields that are mostly used
 internally by MetaMask.
 
-Outside of [Snaps restricted methods](/snaps/reference/rpc-api/#restricted-methods), the only
-restricted method is [`eth_accounts`](/wallet/reference/eth_accounts), which allows you to access
-the user's Ethereum accounts.
-More restricted methods will be added in the future.
+Restricted methods are methods that cannot be called unless you have permission to do so using [`wallet_requestPermissions`](/wallet/reference/wallet_requestpermissions) or [`wallet_requestSnaps`](/wallet/reference/wallet_requestSnaps).
+
+The following methods are restricted:
+
+- [eth_accounts](/wallet/reference/eth_accounts) - Gaining permission requires calling `wallet_requestPermissions`. 
+Granting permissions for `eth_accounts` also provides permissions for the following methods:
+  - [`eth_sendTransaction`](/wallet/reference/eth_sendTransaction)
+  - [`personal_sign`](/wallet/reference/personal_sign)
+  - [`eth_signTypedData_v4`](/wallet/reference/eth_signTypedData_v4)
+
+:::info note
+To access accounts, we recommend using [`eth_requestAccounts`](/wallet/reference/eth_requestAccounts).
+This method automatically obtains permissions for `eth_accounts` through an internal `wallet_requestPermissions` call.
+See [how to access a user's accounts](../how-to/connect/access-accounts.md) for more information.
+:::
+
+- [`wallet_snap`](/wallet/reference/wallet_snap) - Gaining permission requires calling `wallet_requestSnap`.
+- [`wallet_invokeSnap`](/wallet/reference/wallet_invokeSnap) - Gaining permission requires calling `wallet_requestSnap`.
+  
+:::info note
+For more information on requesting permission to connect to `wallet_snap` and `wallet_invokeSnap`, see the example at the end of the [About the Snaps APIs](../../../../snaps/learn/about-snaps/apis.md#custom-json-rpc-apis) page.
```

Comment:
```suggestion
For more information on requesting permission to connect to `wallet_snap` and `wallet_invokeSnap`, see the example at the end of the [About the Snaps APIs](/snaps/learn/about-snaps/apis/#custom-json-rpc-apis) page.
```

Sorry for the confusion, but when linking between `wallet` and `snaps`, you need to use absolute links.